docker/models/containers.py

```diff
@@ -140,6 +140,43 @@ def exec_run(self, cmd, stdout=True, stderr=True, stdin=False, tty=False,
             resp['Id'], detach=detach, tty=tty, stream=stream, socket=socket
         )
 
+    def exec_run2(self, cmd, stdout=True, stderr=True, stdin=False, tty=False,
+                  privileged=False, user='', detach=False,
+                  socket=False):
+        """
+        Run a command inside this container and returns its output and exit
+        code. Similar to ``docker exec``.
+
+        Args:
+            cmd (str or list): Command to be executed
+            stdout (bool): Attach to stdout. Default: ``True``
+            stderr (bool): Attach to stderr. Default: ``True``
+            stdin (bool): Attach to stdin. Default: ``False``
+            tty (bool): Allocate a pseudo-TTY. Default: False
+            privileged (bool): Run as privileged.
+            user (str): User to execute command as. Default: root
+            detach (bool): If true, detach from the exec command.
+                Default: False
+
+        Returns:
+            tuple:
```

Comment:
It might be a good idea to make this return a namedtuple (with `output` and `exit_code`) since then a user could do:
```command = container.exec_run(...)
if command.exit_code == 0:
  ...
```
instead of having to deal with indices.